src/main/java/org/radarbase/management/service/SourceService.java

```diff
@@ -208,24 +212,41 @@ public MinimalSourceDetailsDTO safeUpdateOfAttributes(Source sourceToUpdate,
      * Does not allow to transfer if new project does not have valid source-type.
      *
      * @param sourceDto source details to update.
+     * @param jwt authorization token.
      * @return updated source.
      */
     @Transactional
-    public SourceDTO updateSource(SourceDTO sourceDto) {
-        Source existingSource = sourceRepository.findById(sourceDto.getId()).get();
+    public Optional<SourceDTO> updateSource(SourceDTO sourceDto, RadarToken jwt)
+            throws NotAuthorizedException {
+        Optional<Source> existingSourceOpt = sourceRepository.findById(sourceDto.getId());
+        if (existingSourceOpt.isEmpty()) {
+            return Optional.empty();
+        }
+        Source existingSource = existingSourceOpt.get();
+        String project = existingSource.getProject() != null
+                ? existingSource.getProject().getProjectName()
+                : null;
+        String user = (existingSource.getSubject() != null
+                && existingSource.getSubject().getUser() != null)
+                ? existingSource.getSubject().getUser().getLogin()
+                : null;
+
+        checkPermissionOnSource(jwt, SOURCE_UPDATE, project, user, existingSource.getSourceName());
```

Comment:
Did we already have these methods (checkPermissionOnSource and checkPermissionOnSubject) available in the radar-auth?

Reply:
These have been available quite some time, but they were introduced later than the other permission checks.